internal/wal/wal_aof.go

```diff
@@ -18,43 +20,146 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
-var writeBuf bytes.Buffer
+const (
+	segmentPrefix = "segment-"
+)
+
+var (
+	bufferPool = sync.Pool{
+		New: func() interface{} {
+			return make([]byte, 0, 4096) // Pre-allocate 4KB buffer
+		},
+	}
```

Comment:
we should add this as part of the configuration. The sane default can be 4KB or 8KB.
Also, update the design document with the same.